sky/backends/backend_utils.py

```diff
@@ -1609,17 +1609,19 @@ def _query_status_gcp(
     # TODO(wei-lin): handle multi-node cases.
     if use_tpu_vm and len(status_list) == 0:
         logger.debug(f'Terminating preempted TPU VM cluster {cluster}')
-        # Do not use CloudVMRayBackend.teardown_no_lock, as that will
+        backend = backends.CloudVmRayBackend()
+        handle = global_user_state.get_handle_from_cluster_name(cluster)
+        # Do not use refresh cluster status during teardown, as that will
         # cause inifinite recursion by calling cluster status refresh
         # again.
-        terminate_cmd = tpu_utils.terminate_tpu_vm_cluster_cmd(cluster, zone)
-        returncode, stdout, stderr = log_lib.run_with_log(terminate_cmd,
-                                                          os.devnull,
-                                                          shell=True,
-                                                          stream_logs=False,
-                                                          require_outputs=True)
-        subprocess_utils.handle_returncode(returncode, terminate_cmd,
-                                           stdout + stderr)
+        # The only caller of this function, `_get_cluster_status_via_cloud_cli`,
+        # will do the post teardown cleanup, which will remove the cluster entry
+        # from the status table & the ssh config file.
+        backend.teardown_no_lock(handle,
+                                 terminate=True,
+                                 purge=False,
+                                 post_teardown_cleanup=False,
```

Comment:
If we decide to call this, any reason we should not let it handle cleanup?

Reply:
Good point! I followed the previous logic here, but it makes sense to do the post-teardown cleanup here. I am testing it now to see if that will cause any problem in https://github.com/skypilot-org/skypilot/pull/1555#issuecomment-1365595840